src/rShareTree.ts

```diff
@@ -0,0 +1,153 @@
+        return element;
+    }
+
+    // If a node needs to be collapsible,
+    // change the element condition & return value
+    async getChildren(element?: Node): Promise<Node[]> {
+        if (element) {
+            return;
+        } else {
+            return await this.getNodes();
+        }
+    }
+
+    // To add a tree item to the LiveShare R view,
+    // write a class object that extends Node and
+    // add it to the list of nodes here
+    private async getNodes(): Promise<Node[]> {
+        let items: Node[] = undefined;
+        if (await isLiveShare()) {
+            items = [
+                new ShareNode(),
+                new CommandNode(),
+                new PortNode()
+            ];
+        }
+
+        return items;
+    }
+}
+
+// Base class for adding to
+abstract class Node extends vscode.TreeItem {
+    public label: string;
+    public tooltip: string;
+    public contextValue: string;
+    public description: string;
+    public iconPath: vscode.ThemeIcon;
+    public collapsibleState: vscode.TreeItemCollapsibleState;
+
+    constructor() {
+        super('');
+    }
+}
+
+// Class for any tree item that should have a toggleable state
+// To implement a ToggleNode, in the super, provide a boolean
+// that is used for tracking state.
+// If a toggle is not required, extend a different Node type.
+export abstract class ToggleNode extends Node {
+    public toggle(treeProvider: LiveShareTreeProvider): void { treeProvider.refresh(); }
+    public label: string;
+    public tooltip: string;
+    public contextValue: string;
+    public description: string;
+    public iconPath: vscode.ThemeIcon;
+    public collapsibleState: vscode.TreeItemCollapsibleState;
+
+    constructor(bool: boolean) {
+        super();
+        this.description = bool === true ? 'Enabled' : 'Disabled';
+    }
+
+}
+
+/// Nodes for changing R LiveShare variables
+class ShareNode extends ToggleNode {
+    toggle(treeProvider: LiveShareTreeProvider): void {
+        shareWorkspace = !shareWorkspace;
+        this.description = shareWorkspace === true ? 'Enabled' : 'Disabled';
+        rHostService.orderGuestDetach();
+        treeProvider.refresh();
+    }
+
+    public label: string = 'Share R Workspace';
+    public tooltip: string = 'Whether guests can access the current R session and its workspace';
+    public contextValue: string = 'shareNode';
+    public description: string;
+    public iconPath: vscode.ThemeIcon = new vscode.ThemeIcon('broadcast');
+    public collapsibleState: vscode.TreeItemCollapsibleState = vscode.TreeItemCollapsibleState.None;
+
+    constructor() {
+        super(shareWorkspace);
+    }
+}
+
+class CommandNode extends ToggleNode {
+    toggle(treeProvider: LiveShareTreeProvider): void {
+        forwardCommands = !forwardCommands;
+        this.description = forwardCommands === true ? 'Enabled' : 'Disabled';
+        treeProvider.refresh();
+    }
+
+    public label: string = 'Forward guest commands';
```

Comment:
Instead of `Forward guest commands`, consider making it clear that this is for non-invasive vscode-R commands instead of the R terminal. Possible alternatives are: `Allow guests to interact with R Workspace` or `Allow guests to interact with host R extension`.